Content.Server/Humanoid/Systems/HumanoidSystem.cs

```diff
@@ -116,6 +117,25 @@ public void LoadProfile(EntityUid uid, HumanoidCharacterProfile profile, Humanoi
         }
 
         humanoid.Age = profile.Age;
+        
+        // Caching hair and facial hair colors from their markings
+        if (humanoid.CurrentMarkings.TryGetCategory(MarkingCategories.Hair, out var hairMarkings) &&
+            hairMarkings.Count > 0 &&
+            hairMarkings[0].MarkingColors.Count > 0)
+        {
+            humanoid.CachedHairColor = hairMarkings[0].MarkingColors[0];
```

Comment:
you can just use `FirstOrDefault` here to avoid some of the checks above this and have a default value in place in case there's no markings or that the marking color count is zero